hoursAndWageCalculator.py

```diff
@@ -0,0 +1,178 @@
+        msg = msg + "<td colspan=" + str(6) + "> <b>No registered work time.</b></td>"
+    for employee in employees:
+        empWage = 0
+        name = employee['name']
+        hr = employee['hourly_wage']
+        rates = []
+        for rule in rules:
+            tmp = eval(rule['formula'])
+            rates.append(tmp)
+        normalCoef = rates[1]
+        lateCoef = rates[2]
+        lateLateCoef = rates[3]
+        sundayCoef = rates[0]
+
+        # number of special and normal work hours
+        normalNum = 0
+        lateNum = 0
+        lateLateNum = 0
+        sundayNum = 0
+        # counter is for determining if at least one event exists for a given worker
+        counter = 0
+        for tmpEvent in events:
+            if (-1 != tmpEvent['summary'].find(name) and -1 == tmpEvent['summary'].find(name + "s")):
+                if counter == 0:
+                    counter = 1
+                # only the first 19 chars are used because of timezone formatting
+                startDatetime = datetime.strptime(tmpEvent['start']['dateTime'][:19], '%Y-%m-%dT%H:%M:%S')
+                endDatetime = datetime.strptime(tmpEvent['end']['dateTime'][:19], '%Y-%m-%dT%H:%M:%S')
+                # determining type of work hours
+                # judging by weekday and time
+                tmpNormalNum = 0
+                tmpLateNum = 0
+                tmpLateLateNum = 0
+                tmpSundayNum = 0
+
+                tmpDateTime = startDatetime
+
+                while True:
+
+                    if (dateFunctions.closeEnough(tmpDateTime, endDatetime) == True):
+                        break
+                    holidayInd = dateFunctions.isHoliday(tmpDateTime)
+                    tmpHour = tmpDateTime.hour
+                    if tmpDateTime.minute > 0:
+                        tmpHour+=0.5
+                    if holidayInd is True:
+                        tmpSundayNum += 0.5
+                    elif tmpHour < 8:
+                        tmpLateLateNum += 0.5
+                    elif (tmpHour >= 8 and tmpHour < 17):
+                        tmpNormalNum += 0.5
+                    elif (tmpHour >= 17 and tmpHour < 20):
+                        tmpLateNum += 0.5
+                    elif (tmpHour >= 20):
+                        tmpLateLateNum += 0.5
+                    tmpDateTime = dateFunctions.getNextHalfHourDate(tmpDateTime)
+
+                # WAGE CALCULATOR SUBSECTION
+                wage = normalCoef * tmpNormalNum + lateCoef * tmpLateNum + lateLateCoef * tmpLateLateNum + sundayCoef * tmpSundayNum
+                grandTotal += wage
+                empWage += wage
+                normalNum += tmpNormalNum
+                lateNum += tmpLateNum
+                lateLateNum += tmpLateLateNum
+                sundayNum += tmpSundayNum
+        if counter != 0:
+            msgPrint = msgPrint + (
+                "{:>10}: {:>5}x{} + {:>5}x{} + {:>5}x{} + {:>5}x{} = {:8.2f}\n".format(name, normalNum, normalCoef,
```

Comment:
I kind of dislike this mishmash of code and text. Even the simplest template engine would make this code more elegant, like this one, for example : https://github.com/norvig/pytudes/blob/master/py/yaptu.py

Instead of concatenating strings here, fill some python objects, maps... whatever with the results, and return the result to the caller. 

In this manner, the caller would be able to apply a template engine on the result.

Save the templates for html and text in two separate files.